### PR TITLE
[Quality Of Life] removing programId from metadata of uxd.json

### PR DESCRIPTION
**Pull request by uxd-vincent**
Since `anchor build` or `anchor test` constantly auto-delete the meta-data from the json file, this is kind of a pain to having to constantly manually re-add it.

So in this PR I try to just store the programId in the constants file instead of the IDL file directly

### CODE REVIEW

File: tests/constants.ts [hunk 1]
```diff
@@ -38,7 +37,7 @@ export const CLUSTER = 'devnet';
 
 // ----------------------------------------------------------------------------
 export const uxdProgramId: PublicKey = new PublicKey(
-  jsonIdl['metadata']['address']
+  'EqMrasrCXsbE6mpXaopUH1qip7auwX4AF5UWWpXaLr6u'
```
Comment (cnek) [inline tests/constants.ts:40]: need to take extra care on checking whether this program id matched the one on `./target/deploy/uxd-keypair.json`, but let's go for this changes coz the removal of program id with `anchor build` is indeed super annoying

Comment (uxd-vincent) [reply tests/constants.ts:40]: For sure, True!

Before we had 3 places to double check: lib.rs, uxd.json, Anchor.toml

Now we have 3 places to double check: lib.rs, constants.ts, Anchor.toml

No real improvements here, just avoiding the auto-generation problem that nukes uxd.json

File: .github/workflows/ci-anchor-test.yml [hunk 2]
```diff
@@ -120,6 +120,7 @@ jobs:
           npx @project-serum/anchor-cli@${{ env.ANCHOR_VERSION }} build
           solana program deploy ./target/deploy/uxd.so --program-id ${{ env.RESIDENT_PROGRAM_KEYPAIR }} --upgrade-authority ${{ env.CI_DEPLOYMENT_AUTHORITY_WALLET_PATH }}
       - name: Returning funded runner wallet SOLs to the bank
+        if: always()
```
Comment (cnek) [inline .github/workflows/ci-anchor-test.yml:123]: thx for saving our precious SOL

**Review by cnek — APPROVED** [review]

**Comment by uxd-vincent** [comment]
Getting those errors:

> Error: Metadata property not found in IDL of program: uxd
inside of the Setup Uxd Suite job

It looks like this change might not be possible after all, closing for now, this is not urgent

**Pull request closed without merging**